extensions/amp-social-share/1.0/test/test-social-share.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {SocialShare} from '../social-share';
+import {dict} from '../../../../src/utils/object';
+import {mount} from 'enzyme';
+
+describes.sandboxed('SocialShare preact component v1.0', {}, () => {
+  it('errors when the required "type" attribute is not provided', () => {
+    const jsx = <SocialShare />;
+
+    expect(() => {
+      mount(jsx);
+    }).to.throw('The type attribute is required.');
+  });
+
+  it(
+    'errors when the required endpoint is not provided when not using' +
+      ' a pre-configured type',
+    () => {
+      const props = dict({'type': 'not-configured-type'});
+      const jsx = <SocialShare {...props} />;
+
+      expect(() => {
+        mount(jsx);
+      }).to.throw(
+        'An endpoint is required if not using a pre-configured type.'
+      );
+    }
+  );
+});
```

Comment:
Could any of the tests that use the AMP element have equivalents for the Preact test? I wonder if any cases would be missed when the component is used standalone.

Reply:
That's a good point.  Technically all of the AMP tests can be duplicated onto the Preact side to test the Preact component standalone.  Personally, I feel that the overhead of maintaining these tests may not be worth the potential bugs they may catch as I feel the difference between testing in AMP vs Preact is quite small and there is not that much risk.

Let me know if you feel strongly about it though.

Reply:
Preact tests are essentially a unit test on the component, AMP tests are more or less an integration test on how the Preact components fair in an AMP environment. It's worth having very similar if not identical test cases for both environments because they provide coverage for different things.

Reply:
I actually think having a majority of Preact tests is a good thing (faster). The AMP tests should be specific to code that runs in the `PreactBaseElement` but not in Preact.

Reply:
I agree with this as well.

Reply:
+1 to Justin's comment.

Reply:
To note: we run into an issue with this kind of test: expectation that render function will fail. See https://github.com/ampproject/amphtml/pull/30186